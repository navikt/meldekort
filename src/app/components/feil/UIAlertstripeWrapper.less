.alertstripe_wrapper {
  margin-top: 2rem;
}