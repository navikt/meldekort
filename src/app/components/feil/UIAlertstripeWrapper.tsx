import * as React from 'react';
import { BaksystemFeilmelding } from '../../types/ui';
import AlertStripe from 'nav-frontend-alertstriper';
import { RootState } from '../../store/configureStore';
import { selectFeilmelding } from '../../selectors/ui';
import { Dispatch } from 'redux';
import { UiActions } from '../../actions/ui';
import { connect } from 'react-redux';

interface MapStateToProps {
  baksystemFeilmelding: BaksystemFeilmelding;
}

interface MapDispatchToProps {
  skjulFeilmelding: () => void;
}

type UIAlertstripeWrapperProps = MapStateToProps & MapDispatchToProps;

const UIAlertstripeWrapper: React.FunctionComponent<UIAlertstripeWrapperProps> = ({ baksystemFeilmelding}) => { // { baksystemFeilmelding, skjulFeilmelding }
  return (
    <div className={'alertstripe_wrapper'}>
      {baksystemFeilmelding.visFeilmelding ? (
        <AlertStripe type="feil">
          <div>{baksystemFeilmelding.content()}</div>
        </AlertStripe>
      ) : null}
    </div>
  );
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    baksystemFeilmelding: selectFeilmelding(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => {
  return {
    skjulFeilmelding: () => dispatch(UiActions.skjulBaksystemFeilmelding()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UIAlertstripeWrapper);
