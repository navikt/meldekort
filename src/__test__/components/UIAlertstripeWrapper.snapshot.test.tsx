import * as React from "react";
import * as renderer from "react-test-renderer";
import ProviderWrapper from "../testSetup/providerWrapper";
import UIAlertstripeWrapper from "../../app/components/feil/UIAlertstripeWrapper";

it("UIAlertstripeWrapper", () => {
  const tree = renderer.create(
    <ProviderWrapper>
      <UIAlertstripeWrapper />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});
