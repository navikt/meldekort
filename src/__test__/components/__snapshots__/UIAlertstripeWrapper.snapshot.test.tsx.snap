// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIAlertstripeWrapper 1`] = `
<div
  className="alertstripe_wrapper"
/>
`;
